src/Azure.Functions.Cli/Helpers/PythonHelpers.cs

```diff
@@ -73,65 +73,66 @@ private static void CreateRequirements()
             {
-                return Environment.GetEnvironmentVariable(_pythonDefaultExecutableVar);
+                return new WorkerLanguageVersionInfo(
+                    runtime: WorkerRuntime.python,
+                    executable: Environment.GetEnvironmentVariable(_pythonDefaultExecutableVar));
             }
 
-            const string pythonStr = "python";
-            const string python3Str = "python3";
-            const string python36Str = "python3.6";
-            const string py36Str = "3.6";
-            const string py3Str = "3.";
-            const string warningMessage = "Python 3.6.x is recommended, and used in Azure Functions. You are using Python version {0}.";
-            const string errorIfNotExactMessage = "Python 3.6.x is required, and used in Azure Functions. You are using Python version {0}. "
-                + "Please install Python 3.6, and use a virtual environment to switch to Python 3.6.";
-            const string errorMessageOldPy = "Python 3.x (recommended version 3.6.x) is required. Found python versions ({0}).";
-            const string errorMessageNoPy = "Python 3.x (recommended version 3.6.x) is required. No Python versions were found.";
-
-            var pythonExeVersionTask = VerifyVersion(pythonStr);
-            var python3ExeVersionTask = VerifyVersion(python3Str);
-            var python36ExeVersionTask = VerifyVersion(python36Str);
+            const string warningMessage = "Python 3.6.x or 3.7.x is recommended, and used in Azure Functions. You are using Python version {0}.";
+            const string errorIfNotExactMessage = "Python 3.6.x or 3.7.x is required, and used in Azure Functions. You are using Python version {0}. "
+                + "Please install Python 3.6 or 3.7, and use a virtual environment to switch to Python 3.6 or 3.7.";
+            const string errorMessageOldPy = "Python 3.x (recommended version 3.7.x) is required. Found python versions ({0}).";
+            const string errorMessageNoPy = "Python 3.x (recommended version 3.7.x) is required. No Python versions were found.";
 
-            var pythonExeVersion = await Utilities.SafeExecution(async () => await pythonExeVersionTask) ?? string.Empty;
-            var python3ExeVersion = await Utilities.SafeExecution(async () => await python3ExeVersionTask) ?? string.Empty;
-            var python36ExeVersion = await Utilities.SafeExecution(async () => await python36ExeVersionTask) ?? string.Empty;
-
-            var exeToVersion = new Dictionary<string, string>()
+            var workers = new List<WorkerLanguageVersionInfo>
             {
-                { pythonStr, pythonExeVersion },
-                { python3Str, python3ExeVersion },
-                { python36Str, python36ExeVersion }
+                await GetVersion("python"),
+                await GetVersion("python3"),
+                await GetVersion("python3.6"),
+                await GetVersion("python3.7")
```

Comment:
You can make this parallel like how it was before.

Also, I am not sure if we want to continue with `python3.x` binary version checking.